pkg/thing/payload.go

```diff
@@ -21,51 +21,60 @@ import (
 	"fmt"
 )
 
-// AccessTokenResponse contains the response received from AM after a successful access token request
+// AccessTokenResponse contains the response received from AM after a successful access token request.
+// The response format is specified in https://tools.ietf.org/html/rfc6749#section-4.1.4.
 type AccessTokenResponse struct {
 	Content map[string]interface{}
 }
 
-// AccessToken returns the access token contained in an AccessTokenResponse
+// AccessToken returns the access token contained in an AccessTokenResponse.
 func (a AccessTokenResponse) AccessToken() (string, error) {
 	return a.GetString("access_token")
 }
 
-// ExpiresIn returns the lifetime in seconds of the access token contained in an AccessTokenResponse
+// ExpiresIn returns the lifetime in seconds of the access token contained in an AccessTokenResponse.
 func (a AccessTokenResponse) ExpiresIn() (float64, error) {
 	return a.GetNumber("expires_in")
 }
 
-// GetNumber reads a number from the AccessTokenResponse
+// GetNumber reads a number from the AccessTokenResponse.
 func (a AccessTokenResponse) GetNumber(key string) (float64, error) {
 	if value, ok := a.Content[key].(float64); ok {
 		return value, nil
 	}
 	return 0, errors.New(fmt.Sprintf("failed to read `%s` from response", key))
 }
 
-// GetString reads a string from the AccessTokenResponse
+// GetString reads a string from the AccessTokenResponse.
 func (a AccessTokenResponse) GetString(key string) (string, error) {
 	if value, ok := a.Content[key].(string); ok {
 		return value, nil
 	}
 	return "", errors.New(fmt.Sprintf("failed to read `%s` from response", key))
 }
 
-// AttributesResponse contains the response received from AM after a successful request for attributes
+// AttributesResponse contains the response received from AM after a successful request for thing attributes.
+// The name of the attribute is the same as the LDAP identity attribute name. The response will contain the thing ID
+// and may have multiple values for a single attribute, for example:
+//
+//    {
```

Comment:
Have you checked what this looks like in GoDoc? In the past, I have lost empty lines which can mess with code examples.

Reply:
Yes, I ran this through the godoc tool. It looked fine, but will check once it's picked up by the new online tool.